python/GafferUI/UIEditor.py

```diff
@@ -572,6 +581,55 @@ def __setValue( self, unused, value ) :
 
 		self._updateFromWidget( value )
 
+class _FileSystemPathMetadataWidget( _MetadataWidget ) :
+
+	def __init__( self, key, target = None, acceptEmptyString = True, **kw ) :
+
+		self.__row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal, spacing = 4 )
+		self.__textWidget = GafferUI.TextWidget()
+
+		_MetadataWidget.__init__( self, self.__row, key, target, **kw )
+
+		self.__row.append( self.__textWidget )
+
+		button = GafferUI.Button( image = "pathChooser.png", hasFrame=False )
+		button.clickedSignal().connect( Gaffer.WeakMethod( self.__buttonClicked ), scoped = False )
+		self.__row.append( button )
+
+		self.__acceptEmptyString = acceptEmptyString
+
+		self.__textWidget.editingFinishedSignal().connect(
+			Gaffer.WeakMethod( self.__editingFinished ), scoped = False
+		)
+
+	def textWidget( self ) :
+
+		return self.__textWidget
+
+	def _updateFromValue( self, value ) :
+
+		self.__textWidget.setText( str( value ) if value is not None else "" )
+
+	def __editingFinished( self, *unused ) :
+
+		text = self.__textWidget.getText()
+		if text or self.__acceptEmptyString :
+			self._updateFromWidget( text )
+		else :
+			self._deregisterValue()
+
+	def __buttonClicked( self, widget ) :
+
+		path = self.__textWidget.getText()
+		path = path if os.path.exists( path ) else "/"
```

Comment:
Starting from CWD or HOME is generally friendlier than starting at the root.

Reply:
Addressed in c7f3ff4 as well. Let me know if you want me to drop 23fc005. Just doing the same for `PathPlugValueWidget` there.